### PR TITLE
test setup, pure python implementation, coverage

### PR DESCRIPTION
**Pull request by davisagli**
- add pure-Python implementation
- add MANIFEST.in
- set up tox
- test more environments in travis
- 100% test coverage of Python implementation
- increase next version to 3.0

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -14,26 +14,40 @@
...
+
+
+setup(
+    name='MultiMapping',
+    version='3.0.dev0',
+    url='http://pypi.python.org/pypi/MultiMapping',
+    license='ZPL 2.1',
+    description="Special MultiMapping objects used in Zope2.",
+    author='Zope Foundation and Contributors',
+    author_email='zope-dev@zope.org',
```
Comment (tseaver) [inline setup.py:62]: Needs Trove classifiers for the supported Python versions / implementations.

File: tox.ini [hunk 2]
```diff
@@ -0,0 +1,27 @@
+[tox]
+envlist =
+    py26,py27,py27-pure,py34,pypy,pypy3,coverage
```
Comment (tseaver) [inline tox.ini]: Should include `py32` and `py33`.

**Comment by tseaver** [comment]
Wow, thanks for your work!  Aside from my inline nits, LGTM.

**Comment by davisagli** [comment]
Thanks for catching what I missed. Updated.

**Pull request merged**